### PR TITLE
Run periodic cleanup after installing all packages

### PR DESCRIPTION
**Pull request by FnControlOption**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This makes `brew install` and co. run the monthly automatic cleanup ~before any formulae~ after all packages are installed so that there isn't a wall of text in the middle of the install/upgrade output.

### CODE REVIEW

**Comment by apainintheneck** [comment]
I like this idea but I don't know how necessary it really is since the periodic cleanup only runs periodically. It would be a nicer user experience though because it'd be a bit less confusing.

If we decide to do this, I think there are two things we should consider first.
- We've been discussing running`brew autoremove` after the periodic cleanup based on the presence of the `HOMEBREW_AUTOREMOVE` environment variable (#13532). I mention this because it would make the periodic cleanup output even more verbose. Does that change when we want this to periodic cleanup to happen? Would it be better to run this afterwards instead of before?
- I think it would make more sense for the `#periodic_clean!` and `#install_formula_clean!` methods to be class methods though that would require more refactoring inside of the `Cleanup` class.

**Comment by MikeMcQuaid** [comment]
> * Would it be better to run this afterwards instead of before?

Yeh, I think running the periodic `brew cleanup` after all `brew install`/`brew reinstall`/`brew upgrade` have completed would make more sense than running before (like in this PR) and both make more sense than the current `master` behaviour.

**Comment by FnControlOption** [comment]
Thanks for reviewing, I like what I have now a lot better!!

**Comment by MikeMcQuaid** [comment]
Let's land https://github.com/Homebrew/brew/pull/13532 first, I think it's going to conflict.

**Comment by apainintheneck** [comment]
> Let's land #13532 first, I think it's going to conflict.

@MikeMcQuaid I'm not sure when I'm gonna get around to finishing that PR (things have been busy lately) so go ahead and merge this in when you think it's ready.

**Comment by apainintheneck** [comment]
@FnControlOption I just merged in #13532 which means this should be ready to go after resolving merge conflicts (luckily it doesn't seem like there are a lot). Sorry, it took me so long to finish the other PR.

File: Library/Homebrew/cleanup.rb
Comment (apainintheneck) [inline Library/Homebrew/cleanup.rb:215]: Why has this been changed from `ohai` to `oh1`? It's not that important to me either way; I'm just curious.

**Review by apainintheneck — APPROVED** [review]

File: Library/Homebrew/cleanup.rb
Comment (FnControlOption) [reply Library/Homebrew/cleanup.rb:215]: `periodic_clean!` used to be invoked in place of `install_formula_clean!` but now that it’s invoked on its own, `oh1` makes more sense.

Comment (apainintheneck) [reply Library/Homebrew/cleanup.rb:215]: I see. That makes sense.

**Review by MikeMcQuaid — APPROVED** [review]
Sounds good!

**Comment by MikeMcQuaid** [comment]
Great work again @FnControlOption!

**Pull request merged**
